geode-core/src/main/java/org/apache/geode/internal/cache/ha/HARegionQueue.java

```diff
@@ -2318,7 +2310,7 @@ private int reconcilePutPermits() {
     @Override
     void incrementTakeSidePutPermitsWithoutNotify() {
       synchronized (this.permitMon) {
```

Comment:
look into if this sync is still needed. It was before because all writes of takeSidePutPermits did it while synced on permitMon. But now that you have made takeSidePutPermits atomic and this method does not want to notify it might mean that the sync is no longer needed.

Reply:
The permitmon is what provides the waiting mechanism when the queue is full at 20,000 without it,  it overfills the hrq to 100,000 when the limit is set to 10,000. And the hrq.region.size goes to 200,000 when it should be 20,000. I think we should keep permit mon. All of the other changes look good though.